rpc/src/transport/mod.rs

```diff
@@ -23,7 +24,7 @@ pub mod channel;
 pub trait Transport
 where
     Self: Stream<Item = io::Result<<Self as Transport>::Item>>,
-    Self: Sink<SinkItem = <Self as Transport>::SinkItem, SinkError = io::Error>,
+    Self: Sink<<Self as Transport>::SinkItem, SinkError = io::Error>,
```

Comment:
Unrelated thought: now that Sink is generic over SinkItem, it probably makes sense to make Transport generic over it as well!

I'd accept a separate PR making that change.